### PR TITLE
adding eslint-plugin-graphql as dependency

### PR DESCRIPTION
**Pull request by patsissons**
relates to https://github.com/Shopify/sewing-kit/issues/971 and https://github.com/Shopify/web/pull/6689

This plugin is used by `sewing-kit`. Rather than having sewing kit depend on both this plugin and `eslint-plugin-shopify`, we can instead shuffle `eslint-plugin-graphql` as a dependency of this plugin and pull in all plugins from a single dependency requirement.

### CODE REVIEW

**Comment by lemonmade** [comment]
Presumably we'd want to expose a config for GraphQL that just passes a bunch of stuff through to the ESLint plugin rules?

**Comment by patsissons** [comment]
the base graphql config for the plugin (as per the [online docs](https://github.com/apollographql/eslint-plugin-graphql#example-config-when-using-graphqlconfig))
```js
module.exports = {
  parser: 'babel-eslint',
  rules: {
    "graphql/template-strings": ['error', {
      env: 'literal'
    }]
  },
  plugins: [
    'graphql'
  ]
}
```

**Comment by lemonmade** [comment]
Oh right, it doesn't even need schema info. We can probably just have a new config for `graphql` that does this.

**Comment by patsissons** [comment]
on it! just updating the test

**Comment by patsissons** [comment]
@lemonmade should this be added to the `all` config?

**Comment by lemonmade** [comment]
I think so, IIRC `all` is used for detecting missing configuration for rules

**Comment by patsissons** [comment]
how does that work if the rule from the config is not exposed (like the [`jquery`](https://github.com/Shopify/eslint-plugin-shopify/blob/master/lib/config/jquery.js) config)

**Comment by patsissons** [comment]
In order to test the `all` config, we need to add a `graphql@^0.13.0` devDependency (a [requirement of eslint-plugin-graphql](https://github.com/apollographql/eslint-plugin-graphql/blob/master/package.json#L45)) and supply a dummy `.graphqlconfig.yml` and `schema.graphql` file to enable the plugin to full initialize. Unfortunately, the plugin is not smart enough to bail out gracefully if it is included but no graphql configuration exists.

File: package.json [hunk 1]
```diff
@@ -96,6 +97,7 @@
     "eslint-plugin-react": "7.11.1",
     "eslint-plugin-sort-class-members": "1.3.1",
     "eslint-plugin-typescript": "0.12.0",
+    "graphql": "^0.13.0",
```
Comment (GoodForOneFare) [inline package.json]: `eslint-plugin-graphql` declares a `peerDependency` on graphql.  Isn't that enough?

Comment (patsissons) [reply package.json]: i guess we only need it for testing, i can relegate to a dev dep.

Comment (GoodForOneFare) [reply package.json]: Makes sense, ty 👍

**Review by GoodForOneFare — APPROVED** [review]
Very much appreciated 👍 Thank you for pulling on this thread, then following through so quickly.

**Pull request merged**